collector/jsz.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package collector has various collector utilities and implementations.
+package collector
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	nats "github.com/nats-io/nats-server/v2/server"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	jszSuffix = "/jsz?consumers=true"
```

Comment:
In our call we did say that we make this parametrizable.
The more detail you configure, the more detail you get.
one should have to opt into higher level of detail.

Reply:
sounds good to me, less info by default

Reply:
basically I'd have more options maybe a level and each level adds one more layer. from accounts - streams - consumers
This would translate to using a different suffix. The parsing code you added for the result will be unaltered.